### ISSUE TITLE
Remove support for callbacks in models

### ISSUE THREAD

**Issue by jwerre** [node-oauth/node-oauth2-server#48]
Callback support adds unneeded complexity. Models should be exclusively promise based version ^4.1.0. https://github.com/node-oauth/node-oauth2-server/issues/20#issuecomment-943633108

**Comment by jankapunkt** [comment]
I agree, however

- this will be breaking, thus a new major release imo
- all tests will have to be manually checked against false-positives and false-negatives
- some tests may require to be updated
- if we use promises we can also safely write async functions as they should be natively supported with node >= 12
- then we can also use async functions for async mocha units 
- I still would like to get the release management issue resolved first

**Comment by jwerre** [comment]
@jankapunkt By next major version do you mean v5?  As far as I understand that's  the migration to TypeScript correct? If thats the case it would be nice to do this before all that happens. I think removing support for callbacks can happen now, as we move from **v3** to **v4**. 

We don't really have to do much, just update the documentation and perhaps put a warning message if people try to use callbacks: "support for callbacks has been deprecated—we recommend using Promises in your models since this may have adverse effects in future minor releases". Since we've *officially* removed callback support at v4 and have given people fair warning we should be ok to remove callbacks at any time during the v4 lifecycle.

**Comment by jankapunkt** [comment]
Vice verca this means if we don't break big changes then we can easily do this as 4.1.0

**Comment by jwerre** [comment]
@jankapunkt, @HappyZombies I found some time to work on this a bit today. I removed `bluebird` and `promisify-any` on the `abstract-grant-type.js`. Restricted the tests.

If you get some time take a look and let me know if you have any questions/concerns: https://github.com/node-oauth/node-oauth2-server/commit/4a360eee835feeb6899740d01cf76692b466523e. If all looks good I'll continue to chip away at this.
